### PR TITLE
Close vrepl engine if doing builtin backup with allow_master

### LINKED ISSUE #4973: Backups using -allow_master never completes with custom vreplication
I had a replica get out of sync, so I needed to take a backup from master and restore, a process that typically takes ~10 minutes for this tablet. I ran `vtctlclient -action_timeout="4h0m0s" -server localhost:15999 Backup -allow_master cell-uid` multiple times, increasing the timeout each attempt, and that timed out at 4 hours. I had ~30 vreplication streams live running to the master. After deleting all the vreplication rows, the backup succeeded normally.

### PR DESCRIPTION
**Pull request by deepthi**
vreplication engine needs to be stopped and started if doing an offline backup on MASTER.
Fixes #4973

Signed-off-by: deepthi <deepthi@planetscale.com>

### CODE REVIEW

File: go/vt/vttablet/tabletmanager/rpc_backup.go [hunk 1]
```diff
@@ -83,6 +83,11 @@ func (agent *ActionAgent) Backup(ctx context.Context, concurrency int, logger lo
 			return err
 		}
 		originalType = tablet.Type
+		if originalType == topodatapb.TabletType_MASTER {
```
Comment (sougou) [inline go/vt/vttablet/tabletmanager/rpc_backup.go]: Doesn't the `refreshTablet` call cause the the VREngine to shutdown? I think that's the right way for this to happen.

Comment (deepthi) [reply go/vt/vttablet/tabletmanager/rpc_backup.go]: It doesn't right now, but I agree that is a better place to do this.

Comment (deepthi) [reply go/vt/vttablet/tabletmanager/rpc_backup.go]: oh that's right. This is an old issue and I didn't realize it had been fixed.

**Comment by deepthi** [comment]
Already fixed.

**Pull request closed without merging**